toolkit/cli/commands/src/block_producer_metadata_signatures.rs

```diff
@@ -0,0 +1,103 @@
+use crate::key_params::CrossChainSigningKeyParam;
+use byte_string::ByteString;
+use clap::Parser;
+use parity_scale_codec::Encode;
+use serde::de::DeserializeOwned;
+use serde_json::{self, json};
+use sidechain_domain::*;
+use sp_block_producer_metadata::MetadataSignedMessage;
+use std::io::{BufReader, Read};
+
+#[derive(Clone, Debug, Parser)]
+#[command(author, version, about, long_about = None)]
+pub struct BlockProducerMetadataSignatureCmd {
+	/// Genesis UTXO of the target Partner Chain
+	#[arg(long)]
+	pub genesis_utxo: UtxoId,
+	/// Path of the file containing the metadata in JSON format
+	#[arg(long)]
+	pub metadata_file: String,
+	/// ECDSA signing key of the
+	#[arg(long)]
+	pub cross_chain_signing_key: CrossChainSigningKeyParam,
+}
+
+impl BlockProducerMetadataSignatureCmd {
+	pub fn execute<M: Send + Sync + DeserializeOwned + Encode>(&self) -> anyhow::Result<()> {
```

Comment:
Doesn't `Encode` here makes it problematic to evolve the message in the future? Why not encoding JSON string?